demes/demes.py

```diff
@@ -894,25 +838,41 @@ def deme(
                     "first epoch start time must be less than or equal to "
                     "deme start time"
                 )
-            # set up sizes of first deme, since subsequent demes are built off of it
-            if epochs[0].final_size is None:
-                epochs[0].final_size = epochs[0].initial_size
-            if epochs[0].size_function is None:
-                if epochs[0].initial_size == epochs[0].final_size:
-                    epochs[0].size_function = "constant"
+            # fill in all attributes of epochs
+            for i in range(len(epochs)):
+                # assign default values to epoch attributes if needed
+                if epochs[i].selfing_rate is None:
+                    epochs[i].selfing_rate = selfing_rate
+                if epochs[i].cloning_rate is None:
+                    epochs[i].cloning_rate = cloning_rate
```

Comment:
Is this bit needed here? Do these get set in the `Deme.__attrs_post_init__`?

Reply:
You're right - those are redundant and can be removed.